src/jitlayers.cpp

```diff
@@ -497,6 +499,19 @@ jl_code_instance_t *jl_generate_fptr_impl(jl_method_instance_t *mi JL_PROPAGATES
     return codeinst;
 }
 
+extern "C" JL_DLLEXPORT
+void jl_generate_fptr_for_oc_wrapper_impl(jl_code_instance_t *unspec)
```

Comment:
It still does most of _jl_compile_codeinst, including setting up the JIT, etc., it just emits the LLVM code differently. It also has a different fallback path, since it guarantees that ->specptr is set afterwards, which we can't in the fallback in general. I thought it was akward to try to shoehorn in `jl_generate_fptr_for_unspecialized`.